src/encoded/static/components/genetic_modification.js

```diff
@@ -642,13 +642,28 @@ GeneticModificationComponent.propTypes = {
     context: PropTypes.object.isRequired, // GM object being displayed
     auditIndicators: PropTypes.func.isRequired, // Audit HOC function to display audit indicators
     auditDetail: PropTypes.func.isRequired, // Audit HOC function to display audit details
+    session: PropTypes.object, // Login information from <App>
+};
+
+GeneticModificationComponent.defaultProps = {
+    session: null,
+};
+
+const GeneticModificationInternal = (props, reactContext) => (
+    <GeneticModificationComponent {...props} session={reactContext.session} />
```

Comment:
This looks good, but I was just curious - is there a reason you didn't add `reactContext.session` to `GeneticModificationComponent` and instead made a wrapper?

Reply:
It’s because of a change to React 16 in which componentDidUpdate doesn’t receive information about changing React context anymore. The old componentDidUpdate received a third parameter for the previous React context which I used to see if login information changed. Without that parameter, I had to convert context.session from a React context property to a regular property so componentDidMount could compare that instead, so that new wrapper is there to do that conversion.

Reply:
That makes sense! Thank you!